### DISCUSSION TITLE
Programmatically Toggle Signups

### DISCUSSION THREAD

**Discussion by fangmarks** [supabase/supabase#4015] [Feature Requests]
Basically, it would be great if `Auth` exposed a function that lets people toggle signups if certain criteria are met.

For Example:
```ts
import {supabase} from "./supabase-client"

let {data, error} = await supabase.auth.api.listUsers()

if(data.length === 1) return await supabase.auth.api.signups("disable")

```
Of course, this should only be done server-side, as to not expose the service-role key.

Having a function like this, makes it, IMO, easier for a Developer to make secure but still easy to use websites for people who aren't as Tech-Literate as the Dev.

**Comment by fangmarks** [comment]
I'd personally want a function like this because I want to make something similar to Ghost, but focused on Images instead of Blog posts, but also don't want to force users to Navigate the Supabase Dashboard / Studio Dashboard to disable signups manually after creating their own Account

**Reply by MildTomato** [reply]
Can you elaborate a bit on what you're trying to build?

What do you mean "force users to Navigate the Supabase Dashboard / Studio Dashboard"?

**Reply by fangmarks** [reply]
@MildTomato I'm basically trying to build a gallery site for Artists/Photographers and would love to be able to write functions that'd let me disable signups in Supabase Auth if an ENV like `MAX_SIGNUPS` is reached. 


Some people might not be as Tech-Literate as others so if I can give them a simple GUI, instead of the "more complicated" Dashboard of the Hosted/Self-Hosted Version, to set how many other Uploaders they want to allow on their site, that'd be great
